mediator/med_phases_prep_atm_mod.F90

```diff
@@ -108,15 +108,20 @@ subroutine med_phases_prep_atm(gcomp, rc)
     !--- map atm/ocn fluxes from ocn to atm grid if appropriate
     !---------------------------------------
     if (trim(coupling_mode) == 'cesm' .or. trim(coupling_mode) == 'hafs') then
-       ! Assumption here is that fluxes are computed on the ocean grid
-       call med_map_field_packed( &
-            FBSrc=is_local%wrap%FBMed_aoflux_o, &
-            FBDst=is_local%wrap%FBMed_aoflux_a, &
-            FBFracSrc=is_local%wrap%FBFrac(compocn), &
-            field_normOne=is_local%wrap%field_normOne(compocn,compatm,:), &
-            packed_data=is_local%wrap%packed_data_aoflux_o2a(:), &
-            routehandles=is_local%wrap%RH(compocn,compatm,:), rc=rc)
-       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+       if (is_local%wrap%aoflux_grid == 'ogrid') then
+          call med_map_field_packed( &
+               FBSrc=is_local%wrap%FBMed_aoflux_o, &
+               FBDst=is_local%wrap%FBMed_aoflux_a, &
+               FBFracSrc=is_local%wrap%FBFrac(compocn), &
+               field_normOne=is_local%wrap%field_normOne(compocn,compatm,:), &
+               packed_data=is_local%wrap%packed_data_aoflux_o2a(:), &
+               routehandles=is_local%wrap%RH(compocn,compatm,:), rc=rc)
+          if (ChkErr(rc,__LINE__,u_FILE_u)) return
+       else if (is_local%wrap%aoflux_grid == 'agrid') then
```

Comment:
I am not sure but it might be nice to direct these comments to the mediator log file.